components/bsx/input/TokenBalanceInput.vue

```diff
@@ -7,40 +7,31 @@
     expanded />
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
 import BasicBalanceInput from '@/components/shared/form/BasicBalanceInput.vue'
-import AssetMixin from '@/utils/mixins/assetMixin'
-import { Component, Prop, VModel, mixins } from 'nuxt-property-decorator'
 
-@Component({
-  components: {
-    BasicBalanceInput,
-  },
-})
-export default class TokenBalanceInput extends mixins(AssetMixin) {
-  @VModel() vValue!: string | number
-  @Prop({ type: String, required: false, default: '5' }) tokenId!:
-    | string
-    | number
-  // @Prop({ type: Number, default: '0' }) min!: number
-  // @Prop({ type: Number, default: undefined }) max!: number
+const balanceInputComponent = ref<BasicBalanceInput>()
 
-  get asset() {
-    return this.assetIdOf(this.tokenId)
-  }
+const emit = defineEmits(['input'])
 
-  get unit() {
-    return this.asset.symbol
-  }
+const props = defineProps({
+  value: { type: String || Number, required: true },
+  tokenId: { type: String || Number, required: false, default: '5' },
+})
 
-  get decimals() {
-    return this.asset.decimals
-  }
+const vValue = computed({
+  get() {
+    return props.value
+  },
+  set(value) {
+    emit('input', value)
+  },
+})
 
-  public checkValidity() {
-    const balanceInputComponent = this.$refs
-      .magicBalanceInput as BasicBalanceInput
-    return balanceInputComponent.checkValidity()
-  }
+const asset = computed(() => useAsset().assetIdOf(props.tokenId))
+const unit = computed(() => asset.value.symbol)
+const decimals = computed(() => asset.value.decimals)
+function checkValidity() {
+  return balanceInputComponent.value?.checkValidity()
 }
```

Comment:
Is this even used?
```suggestion
```

Reply:
@roiLeo On the original code, it is not.
I just didn't remove it because it was originally there. And refactoring it shouldn't remove any existing functionalities but rather just improve on it.

Would you like me to remove this?

Reply:
yes please remove, it throw a deepscan issue